sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/csv/CSVPartitionReaderFactory.scala

```diff
@@ -45,7 +45,10 @@ case class CSVPartitionReaderFactory(
     readDataSchema: StructType,
     partitionSchema: StructType,
     parsedOptions: CSVOptions,
-    filters: Seq[Filter]) extends FilePartitionReaderFactory {
+    filters: Seq[Filter])
+  extends FilePartitionReaderFactory(
+    parsedOptions.ignoreCorruptFiles,
+    parsedOptions.ignoreMissingFiles) {
```

Comment:
We can avoid this change, if `CSVOptions` extends `SourceOptions`, and `FilePartitionReaderFactory` simply get these two flags from the options.